### PR TITLE
chore(deps): Update dependencies 2023.01.18

### PR DESCRIPTION
**Pull request by JoseLion**
This PR updates all the dependencies on the library and the example. The most important take of this update is to know that the library work with the latest version of React Native (v0.71.0)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -26,12 +26,14 @@
     "eslint-plugin-import",
     "eslint-plugin-jsdoc",
     "eslint-plugin-react",
+    "etc",
```
Comment (sebas1208) [inline .eslintrc.json:29]: Just wondering what this etc plugin is?

File: src/helpers/common.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import { ReactNode } from "react";
 
+export type OmitR<
+  T extends Record<keyof unknown, unknown>,
+  K extends keyof T,
+> = Pick<T, Exclude<keyof T, K>>;
+
```
Comment (sebas1208) [inline src/helpers/common.ts:7]: Should we document what is the difference between the normal `Omit` type helper and this one?

**Review by sebas1208 — APPROVED** [review]

File: src/helpers/common.ts (hunk 2 above)
Comment (JoseLion) [reply src/helpers/common.ts:7]: Good catch! This one is specifically for Records, so it's good to make it clear on JSDoc 🙂

File: .eslintrc.json (hunk 1 above)
Comment (JoseLion) [reply .eslintrc.json:29]: That one's [eslint-plugin-etc](https://github.com/cartant/eslint-plugin-etc). It provides a bunch of general-purpose TypeScript-specific rules. We use it specifically for two rules:
- `etc/no-commented-out-code`: It disallows code that's been commented while allowing inline comments and JSDoc comments
- `etc/throw-error`: It disallows throwing anything that's not an instance of `Error`

**Pull request merged**